source/jormungandr/jormungandr/parking_space_availability/bss/cykleo.py

```diff
@@ -0,0 +1,147 @@
+    'id': 'cykleo',
+    'name': 'cykleo',
+    'license': 'Private',
+    'url': 'www.cykleo.fr'
+}
+
+
+class CykleoProvider(AbstractParkingPlacesProvider):
+    def __init__(self, url, network, username, password, operators={'cykleo'}, certifi_verify=False,
+                 service_id=None, timeout=2,
+                 feed_publisher=DEFAULT_CYKLEO_FEED_PUBLISHER, **kwargs):
+        self.url = url
+        self.network = network.lower()
+        self.service_id = service_id
+        self.certifi_verify = certifi_verify
+        self.username = username
+        self.password = password
+        self.operators = [o.lower() for o in operators]
+        self.timeout = timeout
+        self.breaker = pybreaker.CircuitBreaker(
+            fail_max=kwargs.get('circuit_breaker_max_fail', app.config['CIRCUIT_BREAKER_MAX_CYKLEO_FAIL']),
+            reset_timeout=kwargs.get('circuit_breaker_reset_timeout', app.config['CIRCUIT_BREAKER_CYKLEO_TIMEOUT_S']))
+        self._feed_publisher = FeedPublisher(**feed_publisher) if feed_publisher else None
+
+    def service_caller(self, method, url, data=None, headers=None):
+        try:
+            kwargs = {"timeout": self.timeout, "verify": self.certifi_verify}
+            if headers:
+                kwargs.update({"headers": headers})
+            if data:
+                kwargs.update({"data": data})
+            response = self.breaker.call(method, url, **kwargs)
+            if not response or response.status_code != 200:
+                logging.getLogger(__name__).error('cykleo, Invalid response, status_code: {}'.format(
+                    response.status_code))
+                return None
+            return response
+        except pybreaker.CircuitBreakerError as e:
+            logging.getLogger(__name__).error('cykleo service dead (error: {})'.format(e))
+        except requests.Timeout as t:
+            logging.getLogger(__name__).error('cykleo service timeout (error: {})'.format(t))
+        except Exception as e:
+            logging.getLogger(__name__).exception('cykleo error : {}'.format(str(e)))
+        return None
+
+    @cache.memoize(app.config['CACHE_CONFIGURATION'].get('TIMEOUT_CYKLEO_JETON', 10*60))
+    def get_access_token(self):
+        headers = {
+            "Content-Type": "application/json",
+            "Accept": "application/json"
+        }
+        data = {"username": self.username, "password": self.password}
+        if self.service_id is None:
+            data.update({"serviceId": self.service_id})
+
+        response = self.service_caller(method=requests.post, url='{}/pu/auth'.format(self.url),
+                                       headers=headers, data=json.dumps(data))
+        if not response:
+            return None
+        content = response.json()
+        access_token = content.get("access_token")
+        if not access_token:
+            logging.getLogger(__name__).error('cykleo, access_token not exist in response')
```

Comment:
English: `cykleo, no access_token in response`

Reply:
Done